pkg/skaffold/runner/dev.go

```diff
@@ -62,32 +61,27 @@ func (r *SkaffoldRunner) doDev(ctx context.Context, out io.Writer) error {
 			color.Default.Fprintf(out, "Syncing %d files for %s\n", len(s.Copy)+len(s.Delete), s.Image)
 
 			if err := r.syncer.Sync(ctx, s); err != nil {
-				r.changeSet.reset()
 				logrus.Warnln("Skipping deploy due to sync error:", err)
 				return nil
 			}
 		}
+	}
 
-	case needsBuild:
+	if needsBuild {
 		defer func() {
 			r.changeSet.resetBuild()
 			r.intents.resetBuild()
 		}()
 
 		if _, err := r.BuildAndTest(ctx, out, r.changeSet.needsRebuild); err != nil {
-			r.changeSet.reset()
```

Comment:
can you explain why we don't need this anymore? don't we want to reset the dirty artifacts so they don't get built again?

Reply:
`r.changeSet.resetBuild()` is called by defer.
There's no reason to reset the rest.

Reply:
@nkubala wdyt?